internal/subaccountsync/cis_accounts.go

```diff
@@ -0,0 +1,76 @@
+package subaccountsync
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+func (c *RateLimitedCisClient) buildSubaccountRequest(subaccountID string) (*http.Request, error) {
+	request, err := http.NewRequest(http.MethodGet, fmt.Sprintf(subaccountServicePath, c.config.ServiceURL, subaccountID), nil)
+	if err != nil {
+		return nil, fmt.Errorf("while creating request: %w", err)
+	}
+	q := request.URL.Query()
+	request.URL.RawQuery = q.Encode()
+	return request, nil
+}
+
+func (c *RateLimitedCisClient) fetchDataForSetOfSubaccounts(subaccounts subaccountsSetType) (statesFromCisType, error) {
+
+	subaccountsDataFromAccounts := make(statesFromCisType)
+	for subaccount := range subaccounts {
+		accountData, err := c.GetSubaccountData(string(subaccount))
+		if err != nil {
+			c.log.Error(fmt.Sprintf("error while getting subaccount data: %s", err))
+		} else {
+			c.log.Debug(fmt.Sprintf("getting for subaccount %s", subaccount))
+			subaccountsDataFromAccounts[subaccount] = accountData
+		}
+	}
+	return subaccountsDataFromAccounts, nil
+}
+
+func (c *RateLimitedCisClient) GetSubaccountData(subaccountID string) (CisStateType, error) {
+	request, err := c.buildSubaccountRequest(subaccountID)
+	if err != nil {
+		return CisStateType{}, fmt.Errorf("while building request for accounts technical service: %w", err)
+	}
+
+	response, err := c.httpClient.Do(request)
```

Comment:
error handling should looks like:
1. check the `err`
2. `defer` statement
3. `NotFound` handling

Reply:
Changed in the next commit.